File: LL/src/Presentation/ll/src/app/shared/models/Dtos/market-place/market-place-listing.ts
import { ItemInstance } from '../../item';

export interface MarketPlaceListing {
  id: string;
  sellerId: string;
  sellerName: string;
  itemInstanceId: string;
  itemInstance: ItemInstance;
  quantity: number;
  unitPrice: number;
  createdAt: Date;
}

File: LL/src/Core/Domain/Models/MarketPlaces/MarketPlaceListing.cs
using Domain.Models.Items;

namespace Domain.Models.MarketPlaces;
public class MarketPlaceListing
{
    public Guid Id { get; set; }
    public Guid SellerId { get; set; }
    public string SellerName { get; set; } = string.Empty;
    public Guid ItemInstanceId { get; set; }
    public ItemInstance ItemInstance { get; set; } = null!;
    public int Quantity { get; set; }
    public long UnitPrice { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset ExpiresAt { get; set; }
}
